core/src/main/java/org/apache/gravitino/policy/PolicyManager.java

```diff
@@ -51,17 +68,51 @@ public PolicyManager(IdGenerator idGenerator, EntityStore entityStore) {
 
   @Override
   public String[] listPolicies(String metalake) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    return Arrays.stream(listPolicyInfos(metalake)).map(Policy::name).toArray(String[]::new);
   }
 
   @Override
   public Policy[] listPolicyInfos(String metalake) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    NameIdentifier metalakeIdent = NameIdentifierUtil.ofMetalake(metalake);
+    return TreeLockUtils.doWithTreeLock(
+        NameIdentifier.of(NamespaceUtil.ofPolicy(metalake).levels()),
+        LockType.READ,
+        () -> {
+          checkMetalake(metalakeIdent, entityStore);
```

Comment:
I think we can move the `checkMetalake` out of lock, it is a low probablitiy that metalake will not exist or not in use when we do policy operations. Moving this out can improve the performance. WDYT?

Reply:
ok, updated